plugins/module_utils/network.py

```diff
@@ -28,6 +28,17 @@ def __init__(self, ip_address, host_mac, switch_mac="00:00:00:00:00:00"):
           'switch_mac': mac_to_bytes(self.switch_mac),
         })
 
+        # find interface matching the specified ip_address
+        interface = None
+        for i in netifaces.interfaces():
+             if i != 'lo':
+                 addr = netifaces.ifaddresses(i)
+                 if netifaces.AF_INET in addr:
+                     for x in addr[netifaces.AF_INET]:
+                         if x['addr'] == ip_address:
+                             interface = i
+                             break
+
```

Comment:
I think this should raise an exception when it does not find the interface.

Having this raising an exception, we do not need `if interface` from the next diff chunk .

Reply:
That's correct indeed. I'll try to come up with a better solution.

This usually happens in case the playbook/inventory does not match to the hosts network settings.